envoy/network/transport_socket.h

```diff
@@ -269,6 +262,21 @@ class TransportSocketFactory {
    * not client-side TLS sockets.
    */
   virtual absl::string_view defaultServerNameIndication() const PURE;
+};
+
+/**
+ * A factory for creating upstream transport sockets. It will be associated to clusters.
+ */
+class TransportSocketFactory : public virtual TransportSocketFactoryBase {
```

Comment:
name this UpstreamTransportSocketFactory?

Reply:
Ack.